src/hooks/web3Context.tsx

```diff
@@ -105,9 +105,15 @@ export const Web3ContextProvider: React.FC<{ children: ReactElement }> = ({ chil
         setTimeout(() => window.location.reload(), 1);
       });
 
-      rawProvider.on("chainChanged", async () => {
+      rawProvider.on("chainChanged", async (_chainId: string) => {
+        const newChainId = idFromHexString(_chainId);
         const networkHash = await initNetworkFunc({ provider });
-        setNetworkId(networkHash.networkId);
+        if (newChainId !== networkHash.networkId) {
+          // then provider is out of sync, reload per metamask recommendation
+          setTimeout(() => window.location.reload(), 1);
+        } else {
+          setNetworkId(networkHash.networkId);
+        }
```

Comment:
I think this is a fix for approve button not firing. In any case it's best practice.